py-vermin: add latest version 1.3.1

netromdk opened the pull request:
New [v1.3.1](https://github.com/netromdk/vermin/releases/tag/v1.3.1) release:
* New rules
* Fixes to older rules
* Bug fixes

@adamjstewart @alalazo

netromdk committed:
py-vermin: add latest version 1.3.1

netromdk commented:
The error found by Vermin via https://github.com/spack/spack/runs/3988990839?check_suite_focus=true#step:5:10 is because some kwarg rules were added especially from py2 to py3. And the `encoding` kwarg of builtin `open()` was added in py 3.0.

For context, in Vermin we added a [wrapper function](https://github.com/netromdk/vermin/blob/master/vermin/utility.py#L194-L199) around the `open()`call.

adamjstewart commented:
Ah, it looks like that line is already wrapped by a Python version check. Can you add `# novm` to ignore that line?

netromdk committed:
Exclude line from Vermin since version is already being checked for

Vermin 1.3.1 finds that `encoding` kwarg of builtin `open()` requires Python 3+.
